File: src/offer4.Search_a_2d_matrix-Array/search_a_2d_matrix.cpp
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        bool find = false;
        int row = array.size();
        int col = array[0].size();
        if( row > 0 && col > 0){
            int r = 0;
            int c = col - 1;
            while(r < row && c >=0){
                if(array[r][c] > target)
                    c--;
                else if(array[r][c] < target)
                    r++;
                else{
                    find = true;
                    break;
                }
            }
        }
        return find;
    }
};

File: src/offer04.Search_a_2d_matrix-Array/search_a_2d_matrix.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        if(array == null || array.length == 0){
            return false;
        }
        int r = array.length;
        int c = array[0].length;
        int i = 0, j = c - 1;
        while(i < r && j >= 0){
            if(array[i][j] == target){
                return true;
            } else if(array[i][j] > target){
                j--;
            } else{
                i++;
            }
        }
        return false;
    }
}